File: servidor/models/producto.py
from sqlalchemy import Column, BigInteger, Text, Integer
from database_dos import Base

class Producto(Base):
    __tablename__ = 'productos'

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    nombre = Column(Text, nullable=False)
    codigo = Column(Text, unique=True, nullable=False)
    talle = Column(Text)
    foto = Column(Text)
    color = Column(Text)
    cantidad_stock_proveedor = Column(Integer, default=0)

File: proveedor/src/main/java/proveedor/models/Producto.java
package proveedor.models;
import lombok.Data;

import jakarta.persistence.*;

@Entity
@Table(name = "productos")
@Data
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "nombre", nullable = false)
    private String nombre;

    @Column(name = "codigo", unique = true, nullable = false)
    private String codigo;

    @Column(name = "talle")
    private String talle;

    @Column(name = "foto")
    private String foto;

    @Column(name = "color")
    private String color;

    @Column(name = "cantidad_stock_proveedor", nullable = false)
    private int cantidadStockProveedor = 0;
}

